### PR TITLE
add i18n option to redirect to first locale

### PR DESCRIPTION
**Pull request by ETLaurent**
## Summary

Add an option to avoid being stuck on a version of a site that has all the locales configured with a prefix, when requesting the site without a locale prefix in the URL.

## What are the specific steps to test this change?

With the following config, http://ca.localhost:3000 should redirect to http://ca.localhost:3000/en-ca and http://localhost:3000 should redirect to http://localhost:3000/es.

```js
      baseUrl: 'http://localhost:3000',
      modules: {
        '@apostrophecms/i18n': {
          options: {
            redirectToFirstLocale: true,
            locales: {
              en: {
                label: 'English',
                hostname: 'en.localhost:3000',
                prefix: '/en'
              },
              'en-CA': {
                label: 'Canada',
                hostname: 'ca.localhost:3000',
                prefix: '/en-ca'
              },
              'fr-CA': {
                label: 'French Canada',
                hostname: 'ca.localhost:3000',
                prefix: '/fr-ca'
              },
              es: {
                label: 'Spain',
                prefix: '/es'
              },
              it: {
                label: 'Italy',
                prefix: '/it'
              }
            }
          }
        }
```

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated --> TODO: https://v3.docs.apostrophecms.org/reference/modules/i18n.html
- [x] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: lib/locales.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ module.exports = {
     ) {
       throw new Error(stripIndent`
         If some of your locales have hostnames, then they all must have
-        hostnames, or your top-level baseUrl option must be set.
+        hostnames, and your top-level baseUrl option must be set.
```
Comment (ETLaurent) [inline lib/locales.js:35]: the `or` was wrong and confusing.
`and` reflects the actual condition

File: test/i18n.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
+const t = require('../test-lib/test.js');
+const assert = require('assert');
+
+describe('static i18n', function() {
```
Comment (ETLaurent) [inline test/i18n.js:4]: existing code moved, starting here ⬇️

File: test/i18n.js [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+  it('should return a 404 HTTP error code when a logged out user tries to access to a content in a private locale', async function() {
+    try {
+      await apos.http.get('/sk');
+    } catch (error) {
+      assert(error.status === 404);
+      return;
+    }
+    throw new Error('should have thrown 404 error');
+  });
+});
```
Comment (ETLaurent) [inline test/i18n.js:136]: existing code moved, ending here ⬆️

File: modules/@apostrophecms/i18n/index.js [hunk 4]
```diff
@@ -202,6 +202,47 @@ module.exports = {
...
+        const localesWithoutHostname = locales.filter(
+          locale => !locale.hostname
+        );
+        const localesWithCurrentHostname = locales.filter(
+          locale => locale.hostname && locale.hostname.split(':')[0] === req.hostname
+        );
+
+        const localesToCheck = localesWithCurrentHostname.length
+          ? localesWithCurrentHostname
+          : localesWithoutHostname;
```
Comment (ETLaurent) [inline modules/@apostrophecms/i18n/index.js:235]: "localesToCheck"... if someone has a better name, I'll take it!

File: test/i18n.js [hunk 5]
```diff
@@ -0,0 +1,310 @@
+const t = require('../test-lib/test.js');
```
Comment (ETLaurent) [inline test/i18n.js:1]: whole file renamed from `test/static-i18n.js` to `test/i18n.js`

**Review by boutell — APPROVED** [review]
Great stuff

**Pull request merged**